### ISSUE TITLE
Add the ability to adjust roll angle to maps

### ISSUE THREAD

**Issue by ehsteve** [sunpy/sunpy#1567]
Sometimes the roll angle in the measurement can be incorrect. Therefore it is helpful to provide the ability to adjust it. The API should be similar to that provided for shift (see #1566).

**Comment by ehsteve** [comment]
@ayshih ping!*

*in lieu of assigning this issue to you.

**Comment by Cadair** [comment]
I feel as though I am standing at the top of a rabbit hole and the whole of wonderland is below my feet!

What if it's not encoded in the header as a angle but as the PC matrix?
What if it's not encoded as PC but CD and we have to account for the pixel scale?
What if the rotation information is not specified at all, in which case we default to a identity matrix?

Also applicable to shift:
How do we save these files back out to a self-consistent FITS header?

**Comment by Cadair** [comment]
This and shift is messing with the immutability of Maps and I am not sure I like it, the system has not been built for this. I think we need to have a serious think about how we handle these kind of operations on a more general level.

**Comment by ehsteve** [comment]
This is related to issue #1549

**Comment by dstansby** [comment]
Is this functionality any different to `my_map.rotate()`?

**Comment by ayshih** [comment]
Yes, it's different.  This is about setting/changing the metadata under the presumption that the current metadata isn't correct for the data.  To be glib and very non-general, this is about providing an API for `.meta['crota2'] = ...`.

`.rotate()` assumes and maintains the consistency of the data and the metadata.

**Comment by wtbarnes** [comment]
Given the recent refactor of the `coalignment` functionality in `sunkit-image` and sunpy/sunkit-image#303, I think I am in favor of having this. I do agree it opens up a can of warms (https://github.com/sunpy/sunpy/issues/1567#issuecomment-144765000), but it is consistent with the `shift_reference_coord` property and is convenient for adjusting incorrect pointing data.
